SimPEG/EM/Static/DC/IODC.py

```diff
@@ -0,0 +1,444 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib
+import properties
+import warnings
+
+import SimPEG
+from SimPEG import Utils, Mesh
+from . import SrcDC as Src   # Pole
+from . import RxDC as Rx
+from .SurveyDC import Survey_ky
+
+
+class IO(properties.HasProperties):
+    """ """
+
+    # Survey
+    survey_geometry = properties.StringChoice(
+        "Survey geometry of DC surveys",
+        default="SURFACE",
+        choices=["SURFACE", "BOREHOLE", "GENERAL"]
+    )
+
+    survey_type = properties.StringChoice(
+        "DC-IP Survey type",
+        default="dipole-dipole",
+        choices=[
+            "dipole-dipole", "pole-dipole",
+            "dipole-pole", "pole-pole"
+        ]
+    )
+
+    dim = properties.Integer(
+        "Dimension of locations",
+        default=2,
+        required=True
+    )
+
+    a_locations = properties.Array(
+        "locations of the positive (+) current electrodes",
+        required=True,
+        shape=('*', '*'),  # ('*', 3) for 3D or ('*', 2) for 2D
+        dtype=float  # data are floats
+    )
+
+    b_locations = properties.Array(
+        "locations of the negative (-) current electrodes",
+        required=True,
+        shape=('*', '*'),  # ('*', 3) for 3D or ('*', 2) for 2D
+        dtype=float  # data are floats
+    )
+
+    m_locations = properties.Array(
+        "locations of the positive (+) potential electrodes",
+        required=True,
+        shape=('*', '*'),  # ('*', 3) for 3D or ('*', 2) for 2D
+        dtype=float  # data are floats
+    )
+
+    n_locations = properties.Array(
+        "locations of the negative (-) potential electrodes",
+        required=True,
+        shape=('*', '*'),  # ('*', 3) for 3D or ('*', 2) for 2D
+        dtype=float  # data are floats
+    )
+
+    electrode_locations = properties.Array(
+        "unique locations of a, b, m, n electrodes",
+        required=True,
+        shape=('*', '*'),  # ('*', 3) for 3D or ('*', 2) for 2D
+        dtype=float  # data are floats
+    )
+
+    # Data
+    data_type = properties.StringChoice(
+        "Type of DC-IP survey",
+        required=True,
+        default="volt",
+        choices=[
+           "volt", "appResistivity", "appConductivity",
+           "voltIP", "appChargeability"
+        ]
+    )
+
+    dobs = properties.Array(
+        "Measured data",
+        shape=('*',),
+        dtype=float  # data are floats
+    )
+
+    volt = properties.Array(
+        "Measured DC voltages (V)",
+        shape=('*',),
+        dtype=float  # data are floats
+    )
+
+    appResistivity = properties.Array(
+        "Measured apparent resistivity (Ohm-m)",
+        shape=('*',),
+        dtype=float  # data are floats
+    )
+
+    appConductivity = properties.Array(
+        "Measured apparent conductivity (S/m)",
+        shape=('*',),
+        dtype=float  # data are floats
+    )
+
+    G = properties.Array(
```

Comment:
I think this should be an `@property`. The properties library is really good for managing things that we want the user to set. If it is something we calculate based on user inputs, it is often better left as an `@property` cc @sgkang

Reply:
@sgkang: would you mind taking a look at this?